File: base_common/src/main/java/com/lazyxu/base/utils/ViewBindingAdapter.java
package com.lazyxu.base.utils;

import androidx.databinding.BindingAdapter;
import android.graphics.drawable.Drawable;
import android.view.View;
import android.widget.ImageView;

import com.bumptech.glide.Glide;

/**
 * User:Lazy_xu
 * Data:2019/10/21
 * Description:
 * FIXME
 */
public class ViewBindingAdapter {

    @BindingAdapter("dbSelected")
    public static void setSelected(View view, boolean selected) {
        view.setSelected(selected);
    }

    @BindingAdapter(value = {"dbImageUrl", "dbError"}, requireAll = false)
    public static void loadImage(ImageView view, String url, Drawable error) {
        Glide.with(view.getContext()).load(url).into(view);
    }

}

File: base_common/src/main/java/com/lazyxu/base/utils/ViewBindingAdapter.kt
package com.lazyxu.base.utils

import android.graphics.drawable.Drawable
import android.text.Editable
import android.text.TextWatcher
import android.view.View
import android.widget.EditText
import android.widget.ImageView
import androidx.databinding.BindingAdapter
import com.bumptech.glide.Glide

/**
 * User:Lazy_xu
 * Data:2019/10/21
 * Description:
 * FIXME
 */
@BindingAdapter("dbSelected")
fun setSelected(view: View, selected: Boolean) {
    view.isSelected = selected
}

@BindingAdapter(value = ["dbImageUrl", "dbError"], requireAll = false)
fun loadImage(view: ImageView, url: String, error: Drawable) {
    Glide.with(view.context).load(url).into(view)
}

@BindingAdapter(value = ["afterTextChanged"])
fun EditText.afterTextChanged(action: (String) -> Unit) {
    addTextChangedListener(object : TextWatcher {
        override fun afterTextChanged(s: Editable?) {
        }

        override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {
        }

        override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
            action(s.toString())
        }
    })
}
